### ISSUE TITLE
CMake Configuration Error: INTERFACE_INCLUDE_DIRECTORIES path is incorrectly prefixed with source directory

### ISSUE THREAD

**Issue by beatlesforever** [aliyun/SimAI#55]
I encountered an error while trying to build SimAI using `./scripts/build.sh -c analytical` .The CMake configuration step fails with the following error:

```
-- Configuring done
CMake Error in src/applications/CMakeLists.txt:
  Target "libapplications" INTERFACE_INCLUDE_DIRECTORIES property contains
  path:

    "/home/zhr/Desktop/SimAI/astra-sim-alibabacloud/extern/network_backend/ns3-interface/simulation/src/applications"

  which is prefixed in the source directory.

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

```

**Comment by Huoyuan100861** [comment]
> I encountered an error while trying to build SimAI using `./scripts/build.sh -c analytical` .The CMake configuration step fails with the following error:
> 
> ```
> -- Configuring done
> CMake Error in src/applications/CMakeLists.txt:
>   Target "libapplications" INTERFACE_INCLUDE_DIRECTORIES property contains
>   path:
> 
>     "/home/zhr/Desktop/SimAI/astra-sim-alibabacloud/extern/network_backend/ns3-interface/simulation/src/applications"
> 
>   which is prefixed in the source directory.
> 
> -- Generating done
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> ```

Could you please provide the detailed information about your machine, such as the OS version, the cmake version, and the gcc/g++ version? Additionally, we'd like to know if you can successfully compile the code on a standard Ubuntu 20.04 system (or higher) or within an NGC container (which we have tested)?

**Comment by beatlesforever** [comment]
Thank you for your response! Here are the details of my machine environment:
1. Operating System:
```
PRETTY_NAME="Ubuntu 22.04.5 LTS"
```

2. CMake Version:
```
cmake version 3.16.3
```

3. GCC/G++ Version:
  ```
  gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0  
  g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0  
  ```

Although I encountered the `INTERFACE_INCLUDE_DIRECTORIES` error during the build process, I was able to successfully compile and run the provided examples afterward. This suggests that the issue might not significantly affect the functionality of the compiled code.

Thank you! 🙏

**Comment by Huoyuan100861** [comment]
> Thank you for your response! Here are the details of my machine environment:
> 
> 1. Operating System:
> 
> ```
> PRETTY_NAME="Ubuntu 22.04.5 LTS"
> ```
> 
> 2. CMake Version:
> 
> ```
> cmake version 3.16.3
> ```
> 
> 3. GCC/G++ Version:
> 
> ```
> gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0  
> g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0  
> ```
> 
> Although I encountered the `INTERFACE_INCLUDE_DIRECTORIES` error during the build process, I was able to successfully compile and run the provided examples afterward. This suggests that the issue might not significantly affect the functionality of the compiled code.
> 
> Thank you! 🙏

Yes, u r right. (Make sure the compilation environment does not include Ninja)

**Comment by riteme** [comment]
I encountered the same problem with CMake 3.30. The compilation stops and complains about ninja. So I removed ninja then re-execute this command and it continues to compile although this error persists.

Requiring ninja not installed is weird. Will it be improved in the future?
